packages/syft/src/syft/core/tensor/autodp/gamma_tensor.py

```diff
@@ -1317,26 +1426,68 @@ def _gt(state: dict) -> jax.numpy.DeviceArray:
 
             output_state[other.id] = other
             child = self.child.__gt__(other.child)
+            output_ds = self.data_subjects > other.data_subjects
 
         else:
 
             def _gt(state: dict) -> jax.numpy.DeviceArray:
                 return jnp.greater(self.run(state), other)
 
             child = self.child.__gt__(other)
+            output_ds = self.data_subjects
 
-        min_val = self.min_val * 0
-        max_val = (self.max_val * 0) + 1
+        min_val = self.min_vals * 0
+        max_val = (self.max_vals * 0) + 1
 
         return GammaTensor(
             child=child,
-            data_subjects=self.data_subjects,
-            min_val=min_val,
-            max_val=max_val,
+            data_subjects=output_ds,
+            min_vals=min_val,
+            max_vals=max_val,
             func=_gt,
             state=output_state,
         )
 
+    def __le__(self, other: Any) -> GammaTensor:
+        # relative
+        from .phi_tensor import PhiTensor
```

Comment:
@madhavajay - we need to figure out a class hierarchy where we don't have any more of these relative imports. In this case we only need it as an argument type checker. What's the right paradigm here? My guess is we need AbstractPhiTensor rand AbstrractGammaTensor to avoid the circular dependency of each tensor needing to check arguments to see if its the other type.